tests/MongoDB.Driver.Examples/CausalConsistencyExamples.cs

```diff
@@ -0,0 +1,124 @@
+                            TimeSpan.FromMilliseconds(1000))
+                    })
+                    .GetCollection<BsonDocument>("items");
+
+                items.UpdateOne(session1,
+                    Builders<BsonDocument>.Filter.And(
+                        Builders<BsonDocument>.Filter.Eq("sku", "111"),
+                        Builders<BsonDocument>.Filter.Eq("end", BsonNull.Value)),
+                    Builders<BsonDocument>.Update.Set("end", currentDate));
+
+                items.InsertOne(session1, new BsonDocument
+                {
+                    {"sku", "nuts-111"},
+                    {"name", "Pecans"},
+                    {"start", currentDate}
+                });
+            }
+            // End Causal Consistency Example 1
+
+            var result = client.GetDatabase(testDatabaseName).GetCollection<BsonDocument>(itemsCollectionName).Find("{}").FirstOrDefault();
+            RemoveIds(new[] { result });
+            result["start"].Should().NotBeNull();
+            result.Remove("start");
+            result.Should().Be("{ \"sku\" : \"nuts-111\", \"name\" : \"Pecans\" }");           
+        }
+
+        [Fact]
+        public void Causal_Consistency_Example_2()
+        {
+            RequireServer.Check().SupportsCausalConsistency();
+
+            string testDatabaseName = "test";
+            string itemsCollectionName = "items";
+            var client = PrepareClient(testDatabaseName, itemsCollectionName);
+
+            using (var session1 = client.StartSession(new ClientSessionOptions { CausalConsistency = true }))
+            {
+                client.GetDatabase(testDatabaseName).RunCommand<BsonDocument>(session1, "{ ping : 1 }");
+
+                // Start Causal Consistency Example 2
+                using (var session2 = client.StartSession(new ClientSessionOptions { CausalConsistency = true }))
+                {
+                    session2.AdvanceClusterTime(session1.ClusterTime);
+                    session2.AdvanceOperationTime(session1.OperationTime);
+
+                    var items = client.GetDatabase("test", new MongoDatabaseSettings
+                        {
+                            ReadPreference = ReadPreference.Secondary,
+                            ReadConcern = ReadConcern.Majority,
+                            WriteConcern = new WriteConcern(WriteConcern.WMode.Majority, TimeSpan.FromMilliseconds(1000))
+                        })
+                        .GetCollection<BsonDocument>("items");
+
+                    var filter = Builders<BsonDocument>.Filter.Eq("end", BsonNull.Value);
+                    foreach (var item in items.Find(session2, filter).ToEnumerable())
+                    {
+                        var print = item;
+                    }
+                }
+                // End Causal Consistency Example 2
+            }
+        }
+
+        private IMongoClient PrepareClient(string database, string collectionName)
```

Comment:
See comment in earlier PR about this method doing two unrelated things. I'd rather see two methods:

1. CreateClient
2. DropCollection